opentracing-spring-extension/opentracing-spring-webflux-extension/src/main/java/org/zalando/opentracing/spring/webflux/extension/ErrorSpanDecorator.java

```diff
@@ -30,7 +30,8 @@ public void onResponse(
             final ServerWebExchange exchange,
             final Span span) {
 
-        if (predicate.test(exchange.getResponse().getStatusCode())) {
+        HttpStatus statusCode = exchange.getResponse().getStatusCode();
+        if (statusCode == null || predicate.test(statusCode)) {
```

Comment:
Under which circumstances is the status null?

Reply:
> Javadoc: The return value may be null if the status code value is outside the HttpStatus enum range, or if there is no default value from the underlying server.

Reply:
it actually happened..

Reply:
I'm not questioning that it happens. But it's important to know the reason, otherwise it's impossible to judge whether null should be considered an error.

Do we have access to the original, low level status as an integer?

Reply:
The javadoc suggests that it's just a normal case where nobody set a status code yet and there is no default. Doesn't sound like an error to me. Neither does a value out of range.

Reply:
HTTP/1.1 mandates presence of status codes, so missing status code seems rather error to me than not

Reply:
there's `getRawStatusCode()` exists, maybe verifying based on it would be better. If not present - error, if present but unknown - not an error.

Reply:
This is a server side filter. I don't think absent status code means what you think it means here.

Reply:
raw status also doesn't help

```java
	@Nullable
	default Integer getRawStatusCode() {
		HttpStatus httpStatus = getStatusCode();
		return (httpStatus != null ? httpStatus.value() : null);
	}
```